azure-mgmt-dns/src/main/java/com/microsoft/azure/management/dns/DnsRecordSet.java

```diff
@@ -138,6 +147,37 @@
         }
 
         /**
+         * The first stage of a CNAME record set definition.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface CNameRecordSetBlank<ParentT> extends WithCNameRecordAlias<ParentT> {
+        }
+
+        /**
+         * The stage of a CNAME record definition allowing to add alias.
+         *
+         * @param <ParentT> the return type of {@link WithAttach#attach()}
+         */
+        interface WithCNameRecordAlias<ParentT> {
+            /**
+             * Creates a CNAME record with the provided alias.
+             *
+             * @param alias the alias
+             * @return the next stage of the definition
+             */
+            WithCNameRecordSetAttachable<ParentT> withAlias(String alias);
```

Comment:
We added a new method in the dns zone `defineCNameRecordSet`. The above method `withAlias` is a nested state of `defineCNameRecordSet`, should we add beta annotation to this instead of for nested state?